src/compiler/checker.ts

```diff
@@ -15034,6 +15034,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             propTypes.push(type);
         }
         addRange(propTypes, indexTypes);
+        if (indexTypes && compilerOptions.noUncheckedIndexedAccess) {

```

Comment:
Maybe I'm just not familiar with this part of the code, but how can this get away with being unconditionally added when this flag is enabled? Are there code paths above this that are missing this too?

Reply:
The preceding branch requires `!indexTypes` and this flag is only relevant to existing `indexTypes`.